doc/content/msr/msre/cardinal_thermal_stripping.md

```diff
@@ -0,0 +1,200 @@
+
+Next, let's examine the case file `tjunc.udf`. The `userq()` function is used to specify the heat source term for temperature calculations or any source terms for the transport equations of passive scalars. In this study, no volumetric heat source is considered for either the fluid or salt regions, so the corresponding source terms are set to zero.
+
+!listing msr/msre/pipe_cardinal/tjunc.udf start=void userq end=void uservp include-end=False
+
+The `uservp()` function defines the material properties for both the fluid and solid regions in conjugate heat transfer simulations. These properties are non-dimensionalized according to nekRS simulation conventions. Consequently, a dimensional conversion is required to transform the raw nekRS results back into dimensional quantities. This conversion is handled automatically by Cardinal, with users simply needing to provide the reference quantities in the input file `nek.i`, which will be discussed in the next section.
+
+!listing msr/msre/pipe_cardinal/tjunc.udf start=void uservp end=void UDF_LoadKernels include-end=False
+
+Given the moderate Reynolds number expected in the T-junction pipe, a LES simulation approach is adopted which can be readily configured in the simulation input file `tjunc.par` using the code snippet below. For more information on Nek5000/nekRS LES filtering, interested readers are referred to [Nek5000 turtorial](https://nek5000.github.io/NekDoc/problem_setup/filter.html). 
+
+```language=bash
+regularization = hpfrt + nModes = 1 + scalingCoeff = 20
+```
+
+As part of the boundary condition configuration, a set of boundary tags is specified in the input parameter file to match the definitions provided in the `usrdat2` subroutine within the `tjunc.usr` file.
+
+!listing msr/msre/pipe_cardinal/tjunc.par start=[VELOCITY] include-end=False
+
+The specific boundary condition values, such as inflow velocities and temperatures, are provided in the case file `tjunc.oudf`. In this setup, the inflow velocity magnitude is set to 1.0 for both the horizontal and vertical inlets, while a non-dimensional temperature of 1.0 is assigned to the vertical inlet.
+
+!listing msr/msre/pipe_cardinal/tjunc.oudf start=void velocityDirichletConditions(bcData *bc) end=@kernel void cliptOKL include-end=False
+
+
+## Cardinal Case Setups
+
+The demonstration model utilizes nekRS to perform conjugate heat transfer simulations and transfer the temperature solution in the solid region (specifically, the T-junction pipe wall) from +nekRS+ to the +MOOSE Solid Mechanics Module+ for thermal stress analysis. This multiphysics coupling approach involves two input files: `cardinal.i` and `nek.i`. In this setup, the MOOSE Solid Mechanics model defined in cardinal.i acts as the master application, while the nekRS CFD model operates as the sub-application.
+Let's first examine the input file `cardinal.i`. A solid region mesh has to be provided in Exodus format, with the mesh cells configured as Hex8 elements.
```

Comment:
how was it generated? do you need to credit any meshing software?

Reply:
the solid mesh of Exodus format was generated by ANSYS ICEM, which is pointed out in the updated documentation.